quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/configuration/mappers/PropertyMapper.java

```diff
@@ -145,9 +154,31 @@ ConfigValue getConfigValue(String name, ConfigSourceInterceptorContext context)
         return transformedValue;
     }
 
-    public Option<T> getOption() { return this.option; }
+    public Option<T> getOption() {
+        return this.option;
+    }
+
+    public void setEnabled(BooleanSupplier enabled) {
+        this.enabled = enabled;
+    }
+
+    public boolean isEnabled() {
+        return enabled.getAsBoolean();
+    }
+
+    public Optional<String> getEnabledWhen() {
+        return Optional.ofNullable(enabledWhen)
```

Comment:
I'm not sure. I quite like this approach as it's clear on the first look that `enabledWhen` is optional and does not always have to be specified. Even though the `isNotBlank` checks the null values, we also need to check whether the string is empty or not. I tend to be a little bit more defensive in these cases to prevent NPE.

I could replace the `isNotBlank` with `.filter(f -> !f.isBlank())` to check only the string value. 

Or what do you suggest?

Reply:
I'd opt for updating the utility method javadoc to specify the null case is handled - and to change it to just str != null && !str.isBlank(). Or if you want directly guard against the null case and use String.isBlank directly, then the utility method should be deprecated / removed.